core/signeddata.go

```diff
@@ -637,6 +637,81 @@ func (r *VersionedSignedValidatorRegistration) UnmarshalJSON(input []byte) error
 	return nil
 }
 
+// NewPartialSignedRandao is a convenience function that returns a new partially signed Randao Reveal.
+func NewPartialSignedRandao(epoch eth2p0.Epoch, sig eth2p0.BLSSignature, shareIdx int) ParSignedData {
+	return ParSignedData{
+		SignedData: SignedRandao{
+			Epoch:        epoch,
+			BLSSignature: sig,
+		},
+		ShareIdx: shareIdx,
+	}
+}
+
+// SignedRandao is a signed Randao Reveal which implements SignedData.
+type SignedRandao struct {
```

Comment:
like issue mentions, please create a SignedEpoch type in eth2util and wrap that type here and make `EpochHashRoot` a method of that type